analyses/bigquery_shim/bigquery_shim/dashboard.py

```diff
@@ -25,6 +25,58 @@ def fetch_results(
         channel_filter = "normalized_channel = '{}' AND".format(channel)
 
     query = """
+    CREATE TEMP FUNCTION hist_to_json(input STRING) AS (
```

Comment:
I wonder if this UDF is generally useful to be added to `mozfun`? Then it would also be easier to add some unit tests.

Reply:
I think a persistent UDF reference may add more complexity than a temp UDF reference, so this may be self-defeating, but I'm not opposed, and testing would be nice. I was hoping to have a single implementation of this logic in `mozfun.hist.extract` such that we don't have to worry about multiple implementations drifting out of sync. But it may be warranted here. I'd propose `mozfun.hist.string_to_json` since `to_json` would imply to me that the input is a parsed `STRUCT` rather than a string.

Reply:
Using persistent UDFs here still keeps the complexity low enough. So, I'll add the `mozfun.hist.string_to_json` to bigquery-etl